[NUI][Visuals] modification of the guide content (#1378)
Samsung/tizen-docs: docs/application/dotnet/guides/nui/visuals.md

1.  Create a visual object of a desirable type:
    ```csharp
    BorderVisual _borderVisual = new BorderVisual();
    ```

2.  Specify the properties required for the visual of that type.
    ```csharp
    // Set the border thickness
    _borderVisual.BorderSize = 10.0f;
    // The border color
    _borderVisual.Color = Color.Blue;
    ```

3.  You can specify the optional properties of a given visual, and also specify the properties that are common for all visual types:
    ```csharp
    // The size of a visual given as a percentage of the size of the parent
    _borderVisual.RelativeSize = new RelativeVector2(0.5f, 0.5f);
    // The parent's reference point
    _borderVisual.Origin = Visual.AlignType.TopBegin;
    // The Visual's reference point
    _borderVisual.AnchorPoint = Visual.AlignType.TopBegin;
    // Shift between Origin and AnchorPoint given as a percentage of a parent size 
    _borderVisual.RelativePosition = new RelativeVector2(0.25f, 0.25f);
    ```

4.  To add a visual to an already existing `VisualView`, use `AddVisual()` as follows:
    ```csharp
    _visualView.AddVisual("name_of_the_visual", _borderVisual);
    ```

## Use VisualView

> `VisualView` is a class that inherits the [Tizen.NUI.BaseComponents.View](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) class. Its properties can be used to specify the `VisualView`.

Reviewer: @mijin85cho Just for learning purposes, where am I wrong in this?

`VisualView` is a class that inherits the [Tizen.NUI.BaseComponents.View](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) class, you can use its properties to specify the `VisualView`.

https://github.com/Samsung/tizen-docs/pull/1378#discussion_r547017651

Reviewer 2: > @mijin85cho Just for learning purposes, where am I wrong in this?
> 
> `VisualView` is a class that inherits the [Tizen.NUI.BaseComponents.View](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) class, you can use its properties to specify the `VisualView`.
> 
> [#1378 (comment)](https://github.com/Samsung/tizen-docs/pull/1378#discussion_r547017651)

@nawab-samsung, as discussed kindly break it or use conjunction.

<!-- TO BE DONE:
(the full list of a `View`'s properties can be found in the [View section](./view.md))
-->

A `VisualView` stores the handles to visuals and maps of their properties. You can find the total number of visuals that are added to the `VisualView` using the `NumberOfVisuals` property. The following are some of the useful methods for working with visuals:

-  To add a visual to the `VisualView`, use the `AddVisual(string, VisualMap)`. The method expects the following two parameters:
   - `string`: specifies the name of the visual, by which it is identified within a given `VisualView`.
   - `VisualMap`: specifies the visual, that is added to the `VisualView`. If the visual with a given name already exists in the `VisualView`, it is updated instead of getting added.

> -  To remove a visual from a `VisualView`, use the `RemoveVisual(string)`. Its only parameter specifies the name of the visual.

Author: there was a comma in your suggestion. great that we agreed now

   _visualView.RelativePosition = new RelativeVector2(0.0f, 0.0f);
   ```

![VisualAlignType_Case1](media/visuals/VisualAlignType_Case1.png)

Case 2:
   ```cs
   _visualView.Origin = Visual.AlignType.TopEnd;
   _visualView.AnchorPoint = Visual.AlignType.Center;
   _visualView.RelativePosition = new RelativeVector2(0.5f, -0.25f);
   ```

![VisualAlignType_Case2](media/visuals/VisualAlignType_Case2.png)

## Visual types provided by NUI

The following visual types are available in NUI:
[Border](#bordervisual),
[Color](#colorvisual),
[Gradient](#gradientvisual),
[Image](#imagevisual),
[NPatch](#npatchvisual),
[SVG](#svgvisual),
[AnimatedImage](#animatedimagevisual),
[Mesh](#meshvisual),
[Primitive](#primitivevisual), and
[Text](#textvisual).

### BorderVisual

The `BorderVisual` renders a rectangular frame with a given thickness. The whole frame is plotted inside the area designated by the `BorderVisual` size as long as it is possible. 

The following examples describe the `BorderVisual` of different sizes: 
- If the `BorderVisual` size is 200x200, and the border width is 100, it appears as a square of the size of 200x200, having the same color as that of the border. 

> - If the `BorderVisual` size is 100x200, and the border width is 200, the frame does not fit inside the given size, and it appears as a rectangle of the size of 300x200, having the same color as that of the frame.

Author: it is correct

| `StartPosition` | `Vector2`                        | For linear only | The start position of the linear gradient. The coordinate system depends on the `Units` value. |
| `EndPosition`   | `Vector2`                        | For linear only | The end position of the linear gradient. The coordinate system depends on the `Units` value.   |
| `Center`        | `Vector2`                        | For radial only | The center point of the radial gradient. The coordinate system depends on the `Units` value.   |
| `Radius`        | `float`                          | For radial only | The size of the radial gradient radius. The coordinate system depends on the `Units` value.    |
| `StopColor`     | `PropertyArray` of `Color`       | Yes             | The color at the stop offsets. At least two colors are required to show a gradient.                 |
| `StopOffset`    | `PropertyArray` of `floats`      | No              | The stop offsets in relative units. To see all the colors from the `StopColor`, the lengths of the `StopColor` and the `StopOffset` arrays must be the same. The default values are `0.0` and `1.0`. |

> | `Units`         | `GradientVisualUnitsType`        | No              | Defines the coordinate system for the attributes: start and end points for a linear gradient, center point and radius for a radial gradient. The default value is `ObjectBoundingBox`. For more information, see the [table](#table_Units_values). |

Reviewer: | `Units`         | `GradientVisualUnitsType`        | No              | Defines the coordinate system for the attributes: start and end points for a linear gradient, center point and radius for a radial gradient. The default value is `ObjectBoundingBox`. For more information, see the [table](#table-units-values). |

Global:
Please verify the links:

Author: I did verify all links in the last commit and they seems to be fine now

> | `SpreadMethod`  | `GradientVisualSpreadMethodType` | No              | Indicates what occurs if a gradient starts or ends inside bounds. The default value is `Pad`. For more information, see the [table](#table_SpreadMethod_values). |

Reviewer: please verify the link!

Author: I did verify all links in the last commit and they seems to be fine now

   _linearGradientVisual.StartPosition = new Vector2(0.0f, 0.5f);
   _linearGradientVisual.EndPosition = new Vector2(0.5f, -0.5f);

   // Optional properties
   _linearGradientVisual.StopColor = new PropertyArray();
   _linearGradientVisual.StopColor.Add(new PropertyValue(Color.Green));
   _linearGradientVisual.StopColor.Add(new PropertyValue(Color.Blue));

   _linearGradientVisual.Opacity = 0.4f;
   _linearGradientVisual.SpreadMethod = GradientVisualSpreadMethodType.Repeat;

   _visualView.AddVisual("Linear_Gradient", _linearGradientVisual);
   ```

The final output after implementing the preceding code is as follows:

 | Radial gradient - Case 1                                    | Radial gradient - Case 2                                    | Linear gradient                                    |
 |:-----------------------------------------------------------:|:-----------------------------------------------------------:|:--------------------------------------------------:|
 | ![](media/visuals/RadialGradientVisualExample_Relative.png) | ![](media/visuals/RadialGradientVisualExample_Absolute.png) | ![](media/visuals/LinearGradientVisualExample.png) |

### ImageVisual

> The `ImageVisual` renders a raster image, such as `.jpg` or `.png` into the control.

Reviewer: ```suggestion
 The `ImageVisual` renders a raster image such as `.jpg` or `.png` into the control.

| `AuxiliaryImageAlpha` | `float`          | No       | An alpha value for mixing between the masked main N-patch image and the auxiliary image.                                                               |
| `ReleasePolicy`     | `ReleasePolicyType`| No       | Specifies whether to release the texture from the cache or to keep it to reduce the loading time. The possible values are `Detached`, `Destroyed`, and `Never`. |
| `LoadPolicy`        | `LoadPolicyType`   | No       | Specifies whether to load the texture immediately after the source is set or only after the visual is added to the window. The possible values are `Immediate` or `Attached`. |
| `OrientationCorrection` | `bool`         | No       | Specifies whether to automatically correct the orientation based on the Exchangeable Image File (EXIF) data. The default value is `true`.              |
| `Atlasing`          | `bool`             | No       | Specifies whether to use the texture atlas or not. The default value is `false`.                                                                       |

> The following example illustrates how to use the `ImageVisual`. The image is placed in an `images` subdirectory under the `res` directory. The absolute path to the application shared resource directory can be resolved by `Tizen.Applications.Application.Current.DirectoryInfo.Resource`. For more information, see [Tizen.Applications.Application](https://samsung.github.io/TizenFX/latest/api/Tizen.Applications.Application.html) and [Tizen.Applications.DirectoryInfo](https://samsung.github.io/TizenFX/latest/api/Tizen.Applications.DirectoryInfo.html):

Reviewer: ```suggestion
The following example illustrates how to use `ImageVisual`. The image is placed in an `images` subdirectory under the `res` directory. The absolute path to the application shared resource directory can be resolved by `Tizen.Applications.Application.Current.DirectoryInfo.Resource`. For more information, see the [Tizen.Applications.Application](https://samsung.github.io/TizenFX/latest/api/Tizen.Applications.Application.html) class and the [Tizen.Applications.DirectoryInfo](https://samsung.github.io/TizenFX/latest/api/Tizen.Applications.DirectoryInfo.html) class:

The following example illustrates how to use the `SVGVisual`:
   ```csharp
   string _imageUrl = Tizen.Applications.Application.Current.DirectoryInfo.Resource + "images/";

   SVGVisual _svgVisual = new SVGVisual();
   _svgVisual.URL = _imageUrl + "tiger.svg";

   _visualView.AddVisual("SVG-Image", _svgVisual);
   ```

The final output after implementing the preceding code, and setting the `_visualView` background color to gray is as follows:

![SVGVisual](media/visuals/SVGVisualExample.png)

### AnimatedImageVisual

The `AnimatedImageVisual` renders an animated image into the control. Currently, only the GIF format is supported.

The following table lists the supported properties:

**Table: AnimatedImageVisual properties.**

| Property            | Type               | Required | Description                                                                                                                 |
|---------------------|--------------------|----------|-----------------------------------------------------------------------------------------------------------------------------|

> | `URL`               | `string`           | Yes, if `URLS` is not specified | The URL of the image.                                                                                |

Author: URLS is correct

| `Octahedron`      | Equivalent to a `BevelledCube` with a `BevelPercentage` of `1`.                      |

Examples of primitives are as follows:

- **Sphere:**

    ![Sphere](media/sphere.png)

- **Conics:**

    | Frustum                                 | Cone                    | Cylinder                        |
    |-----------------------------------------|-------------------------|---------------------------------|
    | ![Frustum](media/conical-frustrum.png)  | ![Cone](media/cone.png) | ![Cylinder](media/cylinder.png) |

- **Bevel:**

   Figures based on different values of the `BevelPercentage` property:

    | 0.0 (cube)                                  | 0.3                                       |
    |---------------------------------------------|-------------------------------------------|
    | ![Cube](media/cube.png)                     | ![Low bevel](media/bevelled-cube-low.png) |
    | **0.7**                                     | **1.0 (octahedron)**                      |
    | ![High bevel](media/bevelled-cube-high.png) | ![Octahedron](media/octahedron.png)       |

- **Slices:**

>     For spheres and conical frustums, `Slices` define how many divisions there are going around the shape.

Reviewer: ```suggestion
 For spheres and conical frustums, `Slices` define how many divisions there are going around the shape:

---
keyword: visual, Border, Color, Gradient, Image, NPatch, SVG, AnimatedImage, Mesh, Primitive, Text, VisualView
---

# Visuals

Visuals are the main building blocks for UI components. They provide reusable rendering logic that is controlled by using properties and can be used by all components. They also respond to view size and color changes and can perform clipping at the renderer level.

> The properties specific to each individual visual are encapsulated in the corresponding class, such as a `Border` visual is defined in a `BorderVisual` class, a `Color` visual in a `ColorVisual` class, and so on. The properties that are common for all visual types are inherited from the [Tizen.NUI.VisualMap](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.VisualMap.html) class. To render a visual, it has to be added to a control. A container class [Tizen.NUI.BaseComponents.VisualView](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.VisualView.html) controls any visual that you have added.

Author: I removed 'individual', but I left the rest unchanged. I think it sounds better this way. Do you agree?

> The properties specific to each individual visual are encapsulated in the corresponding class, such as a `Border` visual is defined in a `BorderVisual` class, a `Color` visual in a `ColorVisual` class, and so on. The properties that are common for all visual types are inherited from the [Tizen.NUI.VisualMap](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.VisualMap.html) class. To render a visual, it has to be added to a control. A container class [Tizen.NUI.BaseComponents.VisualView](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.VisualView.html) controls any visual that you have added.

Reviewer: yes

## Create visual

To create a visual using `BorderVisual` class, follow these steps:

1.  Create a visual object of a desirable type:
    ```csharp
    BorderVisual _borderVisual = new BorderVisual();
    ```

> 2.  Specify the properties required for the visual type. The following section provides detailed information about the properties that are mandatory for a given visual:

Author: I simply removed the second sentence. The idea was to inform the reader, that in the later part of the guide it will be specified which properties are obligatory are which are not.

> 2.  Specify the properties required for the visual type. The following section provides detailed information about the properties that are mandatory for a given visual:

Reviewer 3: @mgrudzinska If so, related with the description in next step, how about adding "mandatory" to this line to specify  that there are mandatory/optional properties?

2.  Specify the mandatory properties required for the visual type. 
...
3. You can specify the optional properties of a given visual, ...
...

> 2.  Specify the properties required for the visual type. The following section provides detailed information about the properties that are mandatory for a given visual:

Author: I thought that the fact, that they are mandatory is clear, since I wrote that they are 'required', but of course I can add it.
now there will be no doubts, that they are mandatory
thanks

    ```csharp
    // Set the border thickness
    _borderVisual.BorderSize = 10.0f;
    // The border color
    _borderVisual.Color = Color.Blue;
    ```

3.  You can specify the optional properties of a given visual, and also specify the properties that are common for all visual types:
    ```csharp
    // The size of a visual given as a percentage of the size of the parent
    _borderVisual.RelativeSize = new RelativeVector2(0.5f, 0.5f);
    // The parent's reference point
    _borderVisual.Origin = Visual.AlignType.TopBegin;
    // The Visual's reference point
    _borderVisual.AnchorPoint = Visual.AlignType.TopBegin;
    // Shift between Origin and AnchorPoint given as a percentage of a parent size 
    _borderVisual.RelativePosition = new RelativeVector2(0.25f, 0.25f);
    ```

4.  To add a visual to an already existing `VisualView`, use `AddVisual()` as follows:
    ```csharp
    _visualView.AddVisual("name_of_the_visual", _borderVisual);
    ```

## Use VisualView

> `VisualView` is a class that inherits the [Tizen.NUI.BaseComponents.View](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) class, you can use its properties to specify the `VisualView`.

Author: please check now

> `VisualView` is a class that inherits the [Tizen.NUI.BaseComponents.View](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) class, you can use its properties to specify the `VisualView`.

Reviewer 3: Thanks for the update, @mgrudzinska

<!-- TO BE DONE:
(the full list of a `View`'s properties can be found in the [View section](./view.md))
-->

A `VisualView` stores the handles to visuals and maps of their properties. You can find the total number of visuals that are added to the `VisualView` using the `NumberOfVisuals` property. The following are some of the useful methods for working with visuals:

-  To add a visual to the `VisualView`, use the `AddVisual(string, VisualMap)`. The method expects the following two parameters:
   - `string`: specifies the name of the visual, by which it is identified within a given `VisualView`.

>    - `VisualMap`: specifies the visual, that will be added to the `VisualView`. If the visual with a given name already exists in the `VisualView`, it will be updated instead of getting added.

Author: fixed, thx

> -  To remove a visual from a `VisualView`, use the `RemoveVisual(string)`, its only parameter specifies the name of the visual.

Author: please check now

> -  To remove a visual from a `VisualView`, use the `RemoveVisual(string)`, its only parameter specifies the name of the visual.

Reviewer 3: Thanks for the update, @mgrudzinska

| `VisualFittingMode` | `VisualFittingModeType`     | `FitKeepAspectRatio` - for AnimatedImageVisual, MeshVisual, PrimitiveVisual, TextVisual<br>`Fill` - otherwise | Specifies the visual fitting mode. For more information, see the [table](#table_VisualFittingMode_possible_values). |

<a name="table_grad_possible_values_units"></a>
**Table: Possible values of the `VisualFittingMode` property.**

| Value                    | Description                                                                                                                       |
|--------------------------|-----------------------------------------------------------------------------------------------------------------------------------|
| `FitKeepAspectRatio`     | The visual is scaled to fit, preserving aspect ratio.                                                                             |
| `Fill`                   | The visual is stretched to fill, not preserving aspect ratio.                                                                     |

### Reference points

> The `Origin` and `AnchorPoint` properties allow determining the position of a visual in the area of the control. The visual and the control are two-dimensional objects, therefore it is necessary to define a reference point on their surfaces for each of them. The following figure illustrates the possible positions of this point, both for the visual and the control:

Author: I removed the part with ''surface', but I'm not sure what else to fix. Could you be more specific please. What is unclear?

> The `Origin` and `AnchorPoint` properties allow determining the position of a visual in the area of the control. The visual and the control are two-dimensional objects, therefore it is necessary to define a reference point on their surfaces for each of them. The following figure illustrates the possible positions of this point, both for the visual and the control:

Reviewer: seems fine now

![VisualAlignTypes](media/visuals/VisualAlignTypes.png)

> The position of the visual is defined as the `AnchorPoint` offset relative to the` Origin` in the reference frame, which is hooked to the upper left corner, and the x and the y-axis are directed to the right and to the bottom respectively. Examples of the various arrangements of the visuals within the control are as follows:

Author: You are right, that it is better to split this sentence into two.
But I think it sounds better with 'TO' in the sentence:
The x and the y-axis are directed to the right and 'TO' the bottom respectively
Do you agree?

> The position of the visual is defined as the `AnchorPoint` offset relative to the` Origin` in the reference frame, which is hooked to the upper left corner, and the x and the y-axis are directed to the right and to the bottom respectively. Examples of the various arrangements of the visuals within the control are as follows:

Reviewer: The position of the visual is defined as the `AnchorPoint` offset relative to the` Origin` in the reference frame, which is hooked to the upper left corner, and the x and the y-axis are directed to the right and to the bottom, respectively. Examples of the various arrangements of the visuals within the control are as follows:

In both cases, the meaning is the same; it is a matter of emphasis. 
Reason: Two prepositional phrases joined by, to, and or may share one preposition. 

When on Monday is joined with on Tuesday, the second on can be dropped: on Monday and Tuesday. The preposition may sometimes be used for added emphasis, but it is generally considered unnecessary and redundant.

| Property        | Type                             | Required        | Description                                                                                    |
|-----------------|----------------------------------|-----------------|------------------------------------------------------------------------------------------------|
| `StartPosition` | `Vector2`                        | For linear only | The start position of the linear gradient. The coordinate system depends on the `Units` value. |
| `EndPosition`   | `Vector2`                        | For linear only | The end position of the linear gradient. The coordinate system depends on the `Units` value.   |
| `Center`        | `Vector2`                        | For radial only | The center point of the radial gradient. The coordinate system depends on the `Units` value.   |
| `Radius`        | `float`                          | For radial only | The size of the radial gradient radius. The coordinate system depends on the `Units` value.    |
| `StopColor`     | `PropertyArray` of `Color`       | Yes             | The color at the stop offsets. At least two are required to show a gradient.                   |

> | `StopOffset`    | `PropertyArray` of `floats`      | No              | The stop offsets in relative units. To see all the colors from the `StopColor`, the lengths of these two arrays should be the same. The default values are `0.0` and `1.0`. |

Reviewer: For more clarity
```suggestion
| `StopOffset`    | `PropertyArray` of `floats`      | No              | The stop offsets in relative units. To see all the colors from the `StopColor`, the length of the `PropertyArray` of `floats`  should be the same. The default values are `0.0` and `1.0`. |
```
Please check if this is technically correct; if no, request you to correct the same.

Author: no, but I changed it. please check it now

> | `Units`         | `GradientVisualUnitsType`        | No              | Defines the coordinate system for the attributes: start and end points for a linear gradient, center point and radius for a radial gradient. The default value is `ObjectBoundingBox`. For more information, see the [table](#table_grad_possible_values_units). |

Author: fixed. sory

| `SpreadMethod`  | `GradientVisualSpreadMethodType` | No              | Indicates what occurs if a gradient starts or ends inside bounds. The default value is `Pad`. For more information, see the [table](#table_grad_possible_values_spread). |

<a name="table_grad_possible_values_units"></a>

> **Table: Possible values of the `Units` property.**

Reviewer: How about  this?

```suggestion
**Table: `Units` properties.** 
```

Just thought to maintain consistency within the doc.

| Value               | Description                                                                                                                                                              |
|---------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `ObjectBoundingBox` | The reference frame in which the top-left corner has the coordinates (-0.5, -0.5) and the bottom-right (0.5, 0.5).                                                       |
| `UserSpace`         | The reference frame with the top-left corner having the coordinates (0, 0) and the coordinates of the bottom-right corner are indicated by the control width and height. |

<a name="table_grad_possible_values_spread"></a>

> **Table: Possible values of the `SpreadMethod` property.**

Reviewer: How about ?

```suggestion
**Table: `SpreadMethod` properties.** 
```

Just thought to maintain consistency within the doc.

| `AlphaMaskURL`      | `string`           | No       | The URL of the alpha mask.                                                                                                                             |
| `AuxiliaryImageURL` | `string`           | No       | The URL of the auxiliary image on top of an N-Patch image.                                                                                             |
| `FittingMode`       | `FittingModeType`  | No       | Fitting options used when resizing the images to fit the specified dimensions. The possible values are `ShrinkToFit`, `ScaleToFill`, `FitWidth`, and `FitHeight`. |
| `SamplingMode`      | `SamplingModeType` | No       | Filtering options used when sampling original pixels to resize images. The possible values are `Box`, `Nearest`, `Linear`, `BoxThenNearest`, `BoxThenLinear`, `NoFilter`, and `DontCare`. |
| `DesiredWidth`      | `int`              | No       | The desired image width. The actual image width is used, if no value is specified.                                                                     |

> | `DesiredHeight`     | `int`              | No       | The desired image height. The actual image height is used, if no value is specified.                                                                   |

Reviewer: ```suggestion
| `DesiredWidth`      | `int`              | No       | The desired width of the image. The actual image width is used, if no value is specified.                                                                     |
| `DesiredHeight`     | `int`              | No       | The desired height of the image. The actual image height is used, if no value is specified.                                                                   |

| `SynchronousLoading`| `bool`             | No       | Specifies whether to load the image synchronously or not. Used for normal quad images only. The default value is `false`.                              |
| `PixelArea`         | `Vector4`          | No       | Specifies the part of the image, that will be displayed. The desired area to be plotted is defined by the relative coordinates of the upper left and the bottom right corners. The default value is `[0.0, 0.0, 1.0, 1.0]`, which implies that the whole image is plotted. |

> | `WrapModeU`<br>`WrapModeV` | `WrapModeType` | No    | Specifies the wrap mode for the U or V coordinate respectively, and how the texture should be sampled when the U or V coordinate exceeds the range of `0.0` to `1.0`. The possible values are `Default`, `ClampToEdge`, `Repeat`, and `MirroredRepeat`. |

Reviewer: ```suggestion
| `PixelArea`         | `Vector4`          | No       | Specifies the part of the image that will be displayed. The desired area to be plotted is defined by the relative coordinates of the upper left and the bottom right corners. The default value is `[0.0, 0.0, 1.0, 1.0]`, which implies that the whole image is plotted. |
| `WrapModeU`<br>`WrapModeV` | `WrapModeType` | No    | Specifies the wrap mode for the U or V coordinate, respectively, and how the texture should be sampled when the U or V coordinate exceeds the range of `0.0` to `1.0`. The possible values are `Default`, `ClampToEdge`, `Repeat`, and `MirroredRepeat`. |

| `MaskContentScale`  | `float`            | No       | Specifies the scale factor to apply to the content image before masking.                                                                               |
| `CropToMask`        | `bool`             | No       | Specifies whether to crop image to mask or scale mask to fit image.                                                                                    |
| `AuxiliaryImageAlpha` | `float`          | No       | An alpha value for mixing between the masked main N-patch image and the auxiliary image.                                                               |
| `ReleasePolicy`     | `ReleasePolicyType`| No       | Specifies whether the texture should be released from the cache or kept to reduce the loading time. The possible values are `Detached`, `Destroyed`, and `Never`. |

> | `LoadPolicy`        | `LoadPolicyType`   | No       | Specifies whether the texture should be loaded immediately after the source set or only after the visual is added to the window. The possible values are `Immediate` or `Attached`. |

Reviewer: ```suggestion
| `LoadPolicy`        | `LoadPolicyType`   | No       | Specifies whether the texture should be loaded immediately after the source is set or only after the visual is added to the window. The possible values are `Immediate` or `Attached`. |

| `OrientationCorrection` | `bool`         | No       | Specifies whether to automatically correct the orientation based on the Exchangeable Image File (EXIF) data. The default value is `true`.              |
| `Atlasing`          | `bool`             | No       | Specifies whether to use the texture atlas or not. The default value is `false`.                                                                       |

> The following example illustrates how to use the `ImageVisual`. The image is placed in an `image` subdirectory under the `res` directory. The absolute path to the application shared resource directory can be resolved by `Tizen.Applications.Application.Current.DirectoryInfo.Resource`. For more information, see [Class Application](https://samsung.github.io/TizenFX/latest/api/Tizen.Applications.Application.html) and [Class DirectoryInfo](https://samsung.github.io/TizenFX/latest/api/Tizen.Applications.DirectoryInfo.html):

Author: right, thx

The following code illustrates how to use the `NPatchVisual`. The absolute path to the image is set as in case of the [`ImageVisual`](#imagevisual):
   ```csharp
   string _imageUrl = Tizen.Applications.Application.Current.DirectoryInfo.Resource + "images/";

   NPatchVisual _nPatchVisual = null;

   _nPatchVisual = new NPatchVisual();
   _nPatchVisual.URL = _imageUrl + "heartsframe.png";
   _nPatchVisual.RelativeSize = new RelativeVector2(1.0f, 0.3f);
   _nPatchVisual.Origin = Visual.AlignType.BottomEnd;
   _nPatchVisual.AnchorPoint = Visual.AlignType.BottomEnd;

   _visualView.AddVisual("NPatch-Image1", _nPatchVisual);

   _nPatchVisual = new NPatchVisual();
   _nPatchVisual.URL = _imageUrl + "heartsframe.png";
   _nPatchVisual.RelativeSize = new RelativeVector2(0.3f, 0.6f);
   _nPatchVisual.Origin = Visual.AlignType.TopCenter;
   _nPatchVisual.AnchorPoint = Visual.AlignType.TopCenter;

   _visualView.AddVisual("NPatch-Image2", _nPatchVisual);
   ```

The final output after implementing the preceding code, and setting the `_visualView` background color to gray is as follows:

![NPatchVisual](media/visuals/NPatchVisualExample.png)

> It is worth paying attention to the correct scaling of the N-Patch image.

Reviewer: ```suggestion 
For correct scaling of the N-Patch image, you must pay due attention.

Author: it was a comment to the above image. Now it does not make sense. I removed this sentence.

| `FrameDelay`        | `float`            | No       | The number of milliseconds between each frame in the visual. The default value is `0.1`.                                    |
| `LoopCount`         | `float`            | No       | The number of times the visual will be looped. For values less than `0` the loop is infinite. The default value is `-1`. |

The following example illustrates how to use the `AnimatedImageVisual`:
   ```csharp
   string _imageUrl = Tizen.Applications.Application.Current.DirectoryInfo.Resource + "images/";

   AnimatedImageVisual _animatedVisual = new AnimatedImageVisual();
   _animatedVisual.URL = _imageUrl + "animated-image-visual.gif";

   _visualView.AddVisual("Animated-Image", _animatedVisual);
   ```

The final output after implementing the preceding code is as follows:

![Animated_Image_Visual](media/animated-image-visual.gif)

### MeshVisual

The `MeshVisual` renders a mesh using a `.obj` file, optionally with materials provided in a `.mtl` file and corresponding textures.

> [!NOTE]

> > Due to a typographical mistake in the source code, it is advised to use 'MaterialtURL' instead of 'MaterialURL' as the property name.

Author: I wanted to use a code style. thanks

|---------------------|----------------------------|--------------------------------------|--------------------------------------------------------------------------------------------------------------------|
| `Shape`             | `PrimitiveVisualShapeType` | all                                  | The shape to render. The default value is `Sphere`. For more information, see the [table](#table_primit_possible_values). |
| `MixColor`          | `Color`                    | all                                  | The color of the shape. The default value is `[0.5f, 0.5f, 0.5f, 1.0f]`.                                           |
| `Slices`            | `int` in a range 1-255     | all                                  | The number of slices going around the shape. The default value is `128`.                                           |
| `Stacks`            | `int`                      | all                                  | The number of layers going down the shape. The default value is `128`.                                             |

> | `ScaleTopRadius`    | `float` greater than or equal to 0 | `ConicalFrustrum`                         | The scale of the top circle radius of a conical frustum. The default value is `1.0`.                  |

Reviewer: | `ScaleTopRadius`    | `float` greater than or equal to 0 | `ConicalFrustrum`                         | The scale of the top circle radius of a conical frustum. The default value is `1.0`.                |

Can you rephrase: The scale of the top circle radius of a conical frustum. The default value is `1.0`. 
Also for : 627,628

Author: Can you specify the reason why it should be rephrased? I think it sounds good so I don't know which part of these sentences to rephrase

Reviewer: @mgrudzinska 
 The scale of the top circle radius of a conical frustum. The default value is 1.0.

This sentence needs to be rephrased, this is not clear. Are these two sentences?

Author: I changed as follows:

The scale of the radius of the top circle of the conical frustum. The default value is `1.0`.

The scale of the radius of the bottom circle of the conic. The default value is `1.5`. 

The scale of the height of the conic. The default value is `3.0`. 

The scale of the radius of the cylinder. The default value is `1.0`.

| `ScaleBottomRadius` | `float` greater than or equal to 0 | `ConicalFrustrum`<br>`Cone`               | The scale of the bottom circle radius. The default value is `1.5`.                                    |
| `ScaleHeight`       | `float` greater than or equal to 0 | `ConicalFrustrum`<br>`Cone`<br>`Cylinder` | The height scale of the conic. The default value is `3.0`.                                            |
| `ScaleRadius`       | `float` greater than or equal to 0 | `Cylinder`                                | The radius scale of the cylinder. The default value is `1.0`.                                         |
| `ScaleDimensions`   | `Vector3`                  | `Cube`<br>`Octahedron`<br>`BevelledCube`          | The dimensions of the cuboid. Scales in the same way as for a 9-patch image. The default value is `[1, 1, 1]`.     |

> | `BevelPercentage`   | `float` in a range 0-1     | `BevelledCube`                       | Specifies how bevelled the cuboid should be, based on the smallest dimension. It affects the ratio of the outer face widths to the width of the overall cube. The default value is `0.0`, which specifies no bevel. |

Reviewer: Can you please elaborate on this :
It affects the ratio of the outer face widths to the width of the overall cube.

Author: please check now

| `BevelSmoothness`   | `float` in a range 0-1     | `BevelledCube`                       | Specifies how smooth the bevelled edges should be. The default value is `0.0`, which specifies sharp edges.        |

> | `LightPosition`     | `Vector3`                  | all                                  | The absolute position of the light source. The point from which the object is illuminated. All zeros indicate the top-left corner in the visual plane. |

Author: please check now
The position of the light source, which illuminates the object.

> | `LightPosition`     | `Vector3`                  | all                                  | The absolute position of the light source. The point from which the object is illuminated. All zeros indicate the top-left corner in the visual plane. |

Reviewer 3: @mgrudzinska Thanks for the suggestion. I think it is better. How about using this description to the `LightPosition` from **MeshVisual properties** table also? In line 582, we have same description.

> | `LightPosition`     | `Vector3`                  | all                                  | The absolute position of the light source. The point from which the object is illuminated. All zeros indicate the top-left corner in the visual plane. |

Author: missed it, sorry

> There are seven `Shape` values, some of which are specializations as compared to another, like a `Cone` is a special case of a `ConicalFrustrum` with `ScaleTopRadius` equal to `0`:

Author: I removed it. this sentence adds nothing

<a name="table_primit_possible_values"></a>

> **Table: Shape values.**

Reviewer: Is the relevant name for the table?

Author: l. 622

Reviewer: What does this mean?

| `ConicalFrustrum` | The area bound between two circles, a cone with the tip removed.                     |
| `Cone`            | Equivalent to a `ConicalFrustrum` with a top radius of 0.                            |
| `Cylinder`        | Equivalent to a `ConicalFrustrum` with equal radii for the top and bottom circles.   |
| `BevelledCube`    | A cube or cuboid with all edges flattened to some degree.                            |
| `Cube`            | Equivalent to a `BevelledCube` with a `BevelPercentage` of `0`.                      |
| `Octahedron`      | Equivalent to a `BevelledCube` with a `BevelPercentage` of `1`.                      |

Examples of primitives are as follows:

- **Sphere:**

    ![Sphere](media/sphere.png)

- **Conics:**

    | Frustum                                 | Cone                    | Cylinder                        |
    |-----------------------------------------|-------------------------|---------------------------------|
    | ![Frustum](media/conical-frustrum.png)  | ![Cone](media/cone.png) | ![Cylinder](media/cylinder.png) |

- **Bevel:**

   Figures based on different values of the `BevelPercentage` property:

>     | 0.0 (cube)                                  | 0.3                                       |

Reviewer: There are no names for 0.3 , 0.7 ??

| `Position`          | `Vector2`                   | `(0.0f, 0.0f)`                   | The visual offset specified as the shift of the visual reference point with respect to the parent's reference point. The values are interpreted as the relative or the absolute, depending on the `PositionPolicy`. |
| `RelativePosition`  | `RelativeVector2`           | `(0.0f, 0.0f)`                   | The visual offset specified as a percentage of the size of the parent.                                            |
| `PositionPolicy`<br>`PositionPolicyX`<br>`PositionPolicyY` | `VisualTransformPolicyType` | `Relative`   | Specifies whether the x and/or y offsets are relative or absolute. The possible values are `Relative` or `Absolute`. |
| `Origin`            | `Visual.AlignType`          | `Center`                         | Specifies the reference point within the parent's area. The possible values are `TopBegin`, `TopCenter`, `TopEnd`, `CenterBegin`, `Center`, `CenterEnd`, `BottomBegin`, `BottomCenter`, and `BottomEnd`. For more information, see [Reference points](#reference_points). |

> | `AnchorPoint`       | `Visual.AlignType`          | `Center`                         | Specifies the reference point within the visual's area. The possible values are `TopBegin`, `TopCenter`, `TopEnd`, `CenterBegin`, `Center`, `CenterEnd`, `BottomBegin`, `BottomCenter`, and `BottomEnd`. For more information, see [Reference points](#reference_points). |

Author: I'm sorry for that. 
I fixed this and checked all others.

|---------------------|--------------------|----------|-----------------------------------------------------------------------------------------------------------------------------|
| `URL`               | `string`           | Yes, if `URLS` is not specified | The URL of the image.                                                                                |
| `URLS`              | `List<string>`     | Yes, if `URL` is not specified  | The list of URLs of the animated images.                                                             |
| `BatchSize`         | `int`              | No       | The batch size for pre-loading images in the visual. The default value is `1`.                                              |
| `CacheSize`         | `int`              | No       | The cache size for loading images in the visual. The default value is `1`.                                                  |
| `FrameDelay`        | `float`            | No       | The number of milliseconds between each frame in the visual. The default value is `0.1`.                                    |

> | `LoopCount`         | `float`            | No       | The number of iterations in a loop. For values less than `0` the loop is infinite. The default value is `-1`.         |

Reviewer: ```suggestion
| `LoopCount`         | `float`            | No       | The number of iterations in a loop. For values less than `0`, the loop is infinite. The default value is `-1`.         |

The following example illustrates how to use the `AnimatedImageVisual`:
   ```csharp
   string _imageUrl = Tizen.Applications.Application.Current.DirectoryInfo.Resource + "images/";

   AnimatedImageVisual _animatedVisual = new AnimatedImageVisual();
   _animatedVisual.URL = _imageUrl + "animated-image-visual.gif";

   _visualView.AddVisual("Animated-Image", _animatedVisual);
   ```

The final output after implementing the preceding code is as follows:

![Animated_Image_Visual](media/animated-image-visual.gif)

### MeshVisual

> The `MeshVisual` renders a mesh using an `.obj` file, optionally with materials provided in an `.mtl` file and textures stored in the provided directory.

Author: I changed it as you suggested to keep the style consistent with other guides.
I used this style in few places, so I changed all '.obj', '.png' etc to OBJ, PNG... 
Thanks

> The `MeshVisual` renders a mesh using an `.obj` file, optionally with materials provided in an `.mtl` file and textures stored in the provided directory.

Reviewer 2: Thanks Mira @mgrudzinska, I understand you have tried to follow the styles in other guides. Thanks for changing PNG as well, I had noticed them but did not explicitly mention to change as I did not want to request too many changes (another reason was that those are commonly used extensions so readers would be familiar).

Really appreciate your effort.

> [!NOTE]
> Due to a typographical mistake in the source code, it is advised to use `MaterialtURL` instead of `MaterialURL` as the property name.

The following table lists the supported properties:

**Table: MeshVisual properties**

| Property         | Type                | Required              | Description                                                                                                                       |
|------------------|---------------------|-----------------------|-----------------------------------------------------------------------------------------------------------------------------------|

> | `ObjectURL`      | `string`            | Yes                   | The location of the `.obj` file.                                                                                                  |

Reviewer 2: If fixing at line 565, kindly fix here as well.

The location of the OBJ file.

> | `MaterialtURL`   | `string`            | No                    | The location of the `.mtl` file. Leave blank for a textureless object.                                                            |

Reviewer 2: If fixing at line 565, kindly fix here as well.

The location of the MTL file.

| `TexturesPath`   | `string`            | Yes, if using materials| The path to the directory where the textures including gloss and normal are stored.                                              |

> | `ShadingMode`    | `MeshVisualShadingModeValue` | No           | The type of the shading mode that the mesh uses. For more information, see the [table](#table_ShadingMode_values).                |

Reviewer: ```suggestion
| `ShadingMode`    | `MeshVisualShadingModeValue` | No           | The type of the shading mode that the mesh uses. For more information, see the [ShadingMode values table](#table_ShadingMode_values). |

| `LightPosition`  | `Vector3`           | No                    | The position of the light source, which illuminates the object. All zeros indicate the top-left corner in the visual plane.       |

The `ShadingMode` property defines the shading mode type used by the mesh.

<a name="table_ShadingMode_values"></a>
**Table: ShadingMode values**

| Enumeration value                      | Description                                                                                                               |
|----------------------------------------|---------------------------------------------------------------------------------------------------------------------------|
| `TexturelessWithDiffuseLighting`       | Uses one color that is lit by ambient and diffuse lighting.                                                               |
| `TexturedWithSpecularLighting`         | Uses only the visual image textures provided with specular lighting in addition to ambient and diffuse lighting.          |
| `TexturedWithDetailedSpecularLighting` | Uses all textures provided including a gloss, normal, and texture map along with specular, ambient, and diffuse lighting. |

> The following example illustrates how to use the `MeshVisual`. The `.obj`, `.mtl` files, and directory with textures use the same path as used for the [`ImageVisual`](#imagevisual):

Reviewer 2: You may have to fix here as well, if you fix above.
The OBJ, MTL files...

   ```csharp
   string _imageUrl = Tizen.Applications.Application.Current.DirectoryInfo.Resource + "images/";

   MeshVisual _meshVisual = new MeshVisual();

   _meshVisual.ObjectURL    = _imageUrl + "Dino.obj";
   _meshVisual.MaterialtURL = _imageUrl + "Dino.mtl";
   _meshVisual.TexturesPath = _imageUrl + "textures/";

   _visualView.AddVisual("Mesh-Image", _meshVisual);
   ```

The final output after implementing the preceding code is as follows:

![Mesh visual](media/visuals/MeshVisualExample.png)

> ## PrimitiveVisual

Author: Thank you for this comment. 
fixed

The `PrimitiveVisual` renders a simple three-dimensional shape, such as cube or sphere. By default, shapes are generated with clockwise winding and back-face culling.

> [!NOTE]
> Due to a typographical mistake in the source code, it is advised to use `ConicalFrustrum` instead of `ConicalFrustum` as one of the `Shape` names. 

The following table lists the supported properties:

**Table: PrimitiveVisual optional properties**

| Property            | Type                       | Shape to which the property applies  | Description                                                                                                        |
|---------------------|----------------------------|--------------------------------------|--------------------------------------------------------------------------------------------------------------------|

> | `Shape`             | `PrimitiveVisualShapeType` | all                                  | The shape to render. The default value is `Sphere`. For more information, see the [table](#table_Shape_values).    |

Reviewer: ```suggestion
| `Shape`             | `PrimitiveVisualShapeType` | all                                  | The shape to render. The default value is `Sphere`. For more information, see the [Shape values table](#table_Shape_values).    |

Examples of primitives are as follows:

- **Sphere:**

    ![Sphere](media/sphere.png)

- **Conics:**

    | Frustum                                 | Cone                    | Cylinder                        |
    |-----------------------------------------|-------------------------|---------------------------------|
    | ![Frustum](media/conical-frustrum.png)  | ![Cone](media/cone.png) | ![Cylinder](media/cylinder.png) |

- **Bevel:**

   Figures based on different values of the `BevelPercentage` property:

    | BevelPercentage = 0.0 (cube)                | BevelPercentage = 0.3                     |
    |---------------------------------------------|-------------------------------------------|
    | ![Cube](media/cube.png)                     | ![Low bevel](media/bevelled-cube-low.png) |
    | **BevelPercentage = 0.7**                   | **BevelPercentage = 1.0 (octahedron)**    |
    | ![High bevel](media/bevelled-cube-high.png) | ![Octahedron](media/octahedron.png)       |

- **Slices:**

    For spheres and conical frustums, `Slices` define how many divisions there are going around the shape:

    ![Slice](media/slices.png)

- **Stacks:**

>     For spheres, `Stacks` defines how many layers there are, going down the shape:

Reviewer: Just remove comma to maintain consistency.
```suggestion 
    For spheres, `Stacks` defines how many layers there are going down the shape:
